core/src/main/java/org/mapfish/print/processor/map/scalebar/ScalebarGraphic.java

```diff
@@ -261,15 +261,18 @@ protected static Dimension getSize(final ScalebarAttributeValues scalebarParams,
                 + settings.getLeftLabelMargin() + settings.getRightLabelMargin();
             height = 2 * settings.getPadding()
                 + settings.getBarSize() + settings.getLabelDistance()
-                + maxLabelSize.height;
+                + (long) Math.ceil(maxLabelSize.height * Math.cos(Math.toRadians(scalebarParams.labelRotation))
+                + maxLabelSize.width * Math.sin(Math.toRadians(scalebarParams.labelRotation)));
         } else {
             width = 2 * settings.getPadding()
-                + maxLabelSize.width + settings.getLabelDistance()
-                + settings.getBarSize();
+                + settings.getLabelDistance() + settings.getBarSize()
+                + (long) Math.ceil(maxLabelSize.width * Math.cos(Math.toRadians(scalebarParams.labelRotation))
+                + maxLabelSize.height * Math.sin(Math.toRadians(scalebarParams.labelRotation)));
             height = 2 * settings.getPadding()
                 + settings.getTopLabelMargin()
                 + settings.getIntervalLengthInPixels() * scalebarParams.intervals
-                + settings.getBottomLabelMargin();
+                + settings.getBottomLabelMargin()
+                + (long) Math.ceil(maxLabelSize.width / 2 * Math.sin(Math.toRadians(scalebarParams.labelRotation)));
```

Comment:
I don't think so... when labelRotation = 0 (default situation), everything stays the same (it is vertical, the labels are printed next to the line and thus do not influence the height)
When there is rotation, the label located at the last tick extends the height (see the four left and right examples for demonstration)
If rotation = 90, the extension is equal to half of the label width (because label is centred at the tick)